service/core.go

```diff
@@ -193,3 +198,14 @@ func (c *Core) nextTaskID() int64 {
 func (c *Core) requestLogger(r *http.Request) *zap.Logger {
 	return c.logger.With(zap.String("request_id", api.RequestIDFromContext(r.Context())))
 }
+
+func (c *Core) publishEvent(event string, data []byte) {
+	payload := fmt.Sprintf("event: %s\ndata: %s\n\n", event, string(data))
```

Comment:
Nits: Might as well do this in the goroutine. And no need to convert `data` to `string` since it'll be formatted by `%s`.